app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
+      vm.model = angular.copy(vm.modelCopy );
+      vm.angularForm.$setUntouched(true);
+      vm.angularForm.$setPristine(true);
+      miqService.miqFlash('warn', __('All changes have been reset'));
+    };
+
+    vm.cancelClicked = function() {
+      if (vm.newRecord) {
```

Comment:
Same here, flash messages and redirects sound like something specific to the form, and not quite related to buttons..